### PR TITLE
ref(theme): Add theme aliases that will be needed for dark mode

### PR DESCRIPTION
**Pull request by billyvg**
This is just a step towards dark mode. Adds aliases to themes that should be used instead of the color directly.

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/theme.tsx [hunk 1]
```diff
@@ -283,6 +366,16 @@ const theme = {
...
 
+  borderRadius: '4px',
+  borderRadiusBottom: '0 0 4px 4px',
+  borderRadiusTop: '4px 4px 0 0',
+  headerSelectorRowHeight: 44,
+  headerSelectorLabelHeight: 28,
+
+  dropShadowLightest: '0 1px 2px rgba(0, 0, 0, 0.04)',
+  dropShadowLight: '0 2px 0 rgba(37, 11, 54, 0.04)',
+  dropShadowHeavy: '0 1px 4px 1px rgba(47,40,55,0.08), 0 4px 16px 0 rgba(47,40,55,0.12)',
```
Comment (billyvg) [inline src/sentry/static/sentry/app/utils/theme.tsx:377]: This is just moved so its not under `colors`

File: src/sentry/static/sentry/app/utils/theme.tsx [hunk 2]
```diff
@@ -67,16 +68,105 @@ const colors = {
...
+
+  /**
+   * A color that denotes a "success", or something good
+   */
+  success: colors.green400, // TODO(dark): colors.green300,
+
+  /**
+   * A color that denotes an error, or something that is wrong
+   */
+  error: colors.red400, // TODO(dark): colors.300,
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/theme.tsx:92]:
```suggestion
  error: colors.red400, // TODO(dark): colors.red300,
```

File: src/sentry/static/sentry/app/utils/theme.tsx [hunk 3]
```diff
@@ -67,16 +68,105 @@ const colors = {
...
+
+  /**
+   * Primary border color
+   */
+  border: colors.gray200,
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/theme.tsx:97]: Do lighter/darker borders need aliases as well?

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/utils/theme.tsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/utils/theme.tsx:97]: The intent (atm) will be to only use a singular border color

Comment (markstory) [reply src/sentry/static/sentry/app/utils/theme.tsx:97]: 🙏

**Review by doralchan — APPROVED** [review]

File: src/sentry/static/sentry/app/utils/theme.tsx [hunk 4]
```diff
@@ -327,7 +420,7 @@ const theme = {
 
   alert,
   badge,
-  button,
+  button: generateButtonTheme(aliases),
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/utils/theme.tsx:423]: This is really cool, but I'm not sure I understand what you're doing very well ... Please, how will you identify if the user has selected a night/light mode and pass this information here? how the aliases with dark colors look like?

Comment (billyvg) [reply src/sentry/static/sentry/app/utils/theme.tsx:423]: Note that this is only a step towards dark mode - the "blueprint" for these changes are at https://github.com/getsentry/sentry/pull/20118 but it involves lots of changes so I'm trying to break them down into digestable chunks.

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**